src/_pytest/pathlib.py

```diff
@@ -99,11 +99,18 @@ def chmod_rw(p: str) -> None:
     func(path)
     return True
 
+def create_long_path(path: Path) -> Path:
```

Comment:
This is a little obscure, so a docstring comment would be helpful IMO. It should explain what problem this solves (can just reference the issue if you want), and also how it solves it, since that's also a little obscure (does `\\?` magically makes it work with long paths? I'm not very familiar with these Windows paths).

Reply:
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation is a pretty good documentation